services/services.go

```diff
@@ -1710,3 +1785,15 @@ func LatestExportedStatisticDay() uint64 {
 	wanted, _ := db.GetLastExportedStatisticDay()
 	return wanted
 }
+
+// LatestTotalSupply will return the last total supply from cache
+func LatestTotalSupply() (string, error) {
+	cacheKey := fmt.Sprintf("%d:frontend:latestTotalSupply", utils.Config.Chain.Config.DepositChainID)
+
+	if totalSupply, err := cache.TieredCache.GetStringWithLocalTimeout(cacheKey, time.Second*5); err == nil {
```

Comment:
Is this protected in case we get the totalSupply from cache before we set it in place? Or generally if the cache key is missing?

Reply:
I follow Beaconcha.in implementation rules: https://github.com/lukso-network/network-explorer-consensus/blob/903b2b1793e83f9e48d2dbd82df251e9a35dbf05/services/services.go#L441 + https://github.com/lukso-network/network-explorer-consensus/blob/903b2b1793e83f9e48d2dbd82df251e9a35dbf05/services/services.go#L1117 so I've assumed that it's preferred design :) But I'm open to discuss